tensorboard/uploader/uploader_main.py

```diff
@@ -90,10 +92,18 @@ def _define_flags(parser):
   subparsers = parser.add_subparsers()
 
   parser.add_argument(
-      '--endpoint',
+      '--origin',
+      type=str,
+      default='https://tensorboard.dev',
+      help='Experimental. Origin for TensorBoard.dev service to which '
+      'to connect, like "https://tensorboard.dev".')
+
+  parser.add_argument(
+      '--api_endpoint',
       type=str,
       default='api.tensorboard.dev:443',
-      help='URL for the API server accepting write requests.')
+      help='Experimental. Direct URL for the API server accepting '
+      'write requests. If set, will skip initial server handshake.')
```

Comment:
See comment on the other PR, but if passing `--origin` in addition to `--api_endpoint` it makes sense to me to still do the server handshake.

Reply:
Will change.

Reply:
Done.